prov/usnic/src/usdf_domain.c

```diff
@@ -285,6 +285,11 @@ usdf_domain_open(struct fid_fabric *fabric, struct fi_info *info,
 		break;
 	case FI_ADDR_STR:
 		sin = usdf_format_to_sin(info, info->src_addr);
+		if (NULL == sin) {
+			free(udp);
```

Comment:
Potential double free situation: 

1. Free `udp` here
2. Jump to `fail`
3. After jumping to `fail` check `if (udp != NULL)` and try to free it again because it's not. 

I see two ways to fix this: 

1. Don't free `udp` here and let it get handled after jumping to the `fail` label.
2. Free it here anyway, but set `udp = NULL` afterwards.